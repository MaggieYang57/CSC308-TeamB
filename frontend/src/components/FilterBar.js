import React, { Component, useState, useEffect } from "react";
import axios from "axios";
import "../css/filterBar.css";
import FilterButton from "./FilterButton";
import SortDrop from "./SortDrop";
import SearchBar from "./SearchBar";

function FilterBar(props) {

  return (
    <div id="filterBar">
      <SearchBar />
      <SortDrop onChange={props.onDiff}/>
      
      <div id="filter-buttons">
        <FilterButton name="Dog Friendly" filterType='dog_friendly' onChange={props.onChange} />
        <FilterButton name="Family Friendly" filterType='family_friendly' onChange={props.onChange}/>
        <FilterButton name="Horseback Riding" filterType='horseback_riding' onChange={props.onChange}/>
        <FilterButton name="Bike Riding" filterType='biking' onChange={props.onChange}/>
      </div>
    </div>
  );
  
}

export default FilterBar;
