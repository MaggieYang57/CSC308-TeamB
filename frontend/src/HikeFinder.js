import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import moment from 'moment';

import { Container, Row, Col } from 'react-bootstrap';
import { HikeCardList } from './components/HikeCardList'
import hikes from './testData';

const averageRatings = (ratings) => {
  let sum = 0
  for (const i in ratings)
    sum += +(ratings[i])
  return (sum / ratings.length).toFixed(1)
}

class AllHikes extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Container >
      <b class="text text-center " style={{fontSize: 50, color: "#2C6674"}}>Hike Finder</b>
      <p class="text text-center " style={{fontSize: 30, color: "#59BCA6"}}>Find one that's just right for you!</p>

      <hr  style={{
          color: '#BDBDBDBD',
          height: 3,
          

      }}/>

        <Col>
          {hikes.map(hike => (
            <Row className="mb-5" key={'${hike.id}'} style={{margin: 20}}>
              <HikeCardList hike={hike} />
            </Row>
          ))}
        </Col>
      </Container>
    );
  }
}
export default AllHikes;
