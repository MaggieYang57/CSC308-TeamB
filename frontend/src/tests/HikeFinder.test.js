import { cleanup } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import HikeFinder from '../HikeFinder';
// eslint-disable-next-line no-unused-vars
import FilterBar from '../components/FilterBar'

jest.mock('../components/FilterBar', () => 'FilterBar');

describe('Hike Finder', () => {

  beforeEach(() => {

  });

  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter> <HikeFinder hikeList={[]} /> </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
