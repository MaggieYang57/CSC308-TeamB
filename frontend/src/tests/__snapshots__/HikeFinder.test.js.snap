// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hike Finder renders correctly 1`] = `
Array [
  " ",
  <FilterBar
    onChange={[Function]}
    onDiff={[Function]}
    updateBySearch={[Function]}
  />,
  <div
    className="container"
    style={
      Object {
        "marginTop": "5vw",
      }
    }
  >
    <b
      className="text text-center "
      style={
        Object {
          "color": "#2C6674",
          "fontSize": 50,
        }
      }
    >
      Hike Finder
    </b>
    <p
      className="text text-center "
      style={
        Object {
          "color": "#59BCA6",
          "fontSize": 30,
        }
      }
    >
      Find one that is just right for you!
    </p>
  </div>,
  <div
    className="container"
    style={
      Object {
        "marginTop": "2vw",
      }
    }
  >
    <hr
      style={
        Object {
          "color": "#BDBDBDBD",
          "height": 3,
        }
      }
    />
  </div>,
  <div
    className="container"
  >
    <div
      className="col"
    />
  </div>,
  " ",
]
`;
